src/hangar/dataset.py

```diff
@@ -670,68 +755,109 @@ def init_dataset(self, name: str, shape=None, dtype=None, prototype=None, *,
         shape : tuple, optional
             The shape of the data samples which will be written in this dataset.
             This argument and the `dtype` argument are required if a `prototype`
-            is not provided, defaults to None
+            is not provided, defaults to None.
         dtype : np.dtype, optional
             The datatype of this dataset. This argument and the `shape` argument
-            are required if a `prototype` is not provided., defaults to None
+            are required if a `prototype` is not provided., defaults to None.
         prototype : np.array, optional
             A sample array of correct datatype and shape which will be used to
             initialize the dataset storage mechanisms. If this is provided, the
-            `shape` and `dtype` arguments must not be set, defaults to None
+            `shape` and `dtype` arguments must not be set, defaults to None.
         samples_are_named : bool, optional
             If the samples in the dataset have names associated with them. If set,
             all samples must be provided names, if not, no name will be assigned.
             defaults to True, which means all samples should have names.
         variable_shape : bool, optional
             If this is a variable sized dataset. If true, a `max_shape` argument
-            must be specified, defaults to False
+            must be specified, defaults to False.
         max_shape : tuple of int, optional
             The maximum size for each dimension which a data sample can be set
             with. The number of dimensions must match that specified in the
             `shape` or `prototype` argument, and each dimension size must be >=
-            the equivalent dimension size specified. defaults to None
+            the equivalent dimension size specified. defaults to None.
 
         Returns
         -------
-        object
+        :class:``DatasetDataWriter``
             instance object of the initialized dataset.
 
+        Raises
+        ------
+        SyntaxError
+            If provided name contains any non ascii, alpha-numeric characters.
+        SyntaxError
+            If required `shape` and `dtype` arguments are not provided in absense of
+            `prototype` argument.
+        LookupError
+            If a dataset already exists with the provided name.
+        ValueError
+            If provided prototype shape (or `shape` argument) not <= `max_shape`
+            value if `variable_shape=True`.
+        ValueError
+            If rank of maximum tensor shape > 31.
         '''
-        if not (name.isalnum or name.isascii):
-            print(f'dataset name provided: {name} is not allowed. Must only contain '
-                  f'alpha-numeric ascii characters with no whitespace characters.')
-            return False
 
         # ------------- Checks for argument validity --------------------------
 
+        if not (name.isalnum or name.isascii):
```

Comment:
- isalnum is a function and needs to be called
- isascii is a new feature came in 3.7 and won't work with 3.6

I have a half cooked PR with this fix and some others at #15

Reply:
Great. i've fixed them here and will merge these so they are one logical set of additions.